lib/Sema/TypeCheckType.cpp

```diff
@@ -1400,23 +1400,18 @@ static bool diagnoseAvailability(Type ty, IdentTypeRepr *IdType, SourceLoc Loc,
   return false;
 }
 
-/// Whether the given DC is a noescape-by-default context, i.e. not a property
-/// setter
-static bool isDefaultNoEscapeContext(const DeclContext *DC) {
-  auto funcDecl = dyn_cast<FuncDecl>(DC);
-  return !funcDecl || !funcDecl->isSetter();
-}
-
 // Hack to apply context-specific @escaping to an AST function type.
-static Type applyNonEscapingFromContext(DeclContext *DC,
-                                        Type ty,
-                                        TypeResolutionOptions options) {
+static Type applyNonEscapingFromContext(DeclContext *DC, Type ty,
+                                        TypeResolutionOptions options,
+                                        bool isSetterNewValue) {
   // Remember whether this is a function parameter.
   bool isFunctionParam =
```

Comment:
Why don't you just pass in isFunctionParam instead of passing in options and isSetterNewValue